orttraining/orttraining/training_ops/cuda/communication/nccl_service.cc

```diff
@@ -233,6 +233,9 @@ void NcclService::Initialize() {
   // Set device this NCCL communicator runs on.
   CUDA_CALL(cudaSetDevice(mpi_rank));
 
+  // Create communication stream.
+  CUDA_CALL(cudaStreamCreate(&stream_));
```

Comment:
This new stream will be implicitly synced with the default stream. Does the current pipeline implementation depend on this implicit sync? If not, then you can create a unblocking stream as below.

cudaStreamCreateWithFlags(&stream_, cudaStreamNonBlocking)

Reply:
No. I will use your suggestion. Thanks.